src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -327,39 +327,20 @@ internal NpgsqlConnector(ConnectorSource connectorSource, NpgsqlConnection conn)
             State = ConnectorState.Closed;
             TransactionStatus = TransactionStatus.Idle;
             Settings = connectorSource.Settings;
+            SslMode = Settings.SslMode;
             PostgresParameters = new Dictionary<string, string>();
 
             CancelLock = new object();
 
             _isKeepAliveEnabled = Settings.KeepAlive > 0;
-            if (_isKeepAliveEnabled)
-            {
-                _userLock = new SemaphoreSlim(1, 1);
-                _keepAliveTimer = new Timer(PerformKeepAlive, null, Timeout.Infinite, Timeout.Infinite);
-            }
+            SetupKeepAlive();
```

Comment:
These are only called from this constructor, any specific reason to extract them out to a function? I'd keep the code inline, or at least in a local method here...

Reply:
🤦 missed that thing. They should be called just before the next loop for `SSLMode.Disabled`.